src/test/java/io/jenkins/plugins/analysis/core/graphs/SeriesBuilderTest.java

```diff
@@ -0,0 +1,317 @@
+package io.jenkins.plugins.analysis.core.graphs;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.assertj.core.util.Lists;
+import org.jfree.data.category.CategoryDataset;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import static io.jenkins.plugins.analysis.core.graphs.assertj.Assertions.*;
+import io.jenkins.plugins.analysis.core.quality.AnalysisBuild;
+import io.jenkins.plugins.analysis.core.quality.StaticAnalysisRun;
+import static java.util.Arrays.*;
+import static org.mockito.Mockito.*;
+
+
+class SeriesBuilderTest {
+
+
+    private static final GraphConfiguration CONFIG_BUILD_COUNT_NONE = createWithBuildCount(0);
+    private static final GraphConfiguration CONFIG_BUILD_COUNT_ONE = createWithBuildCount(1);
+    private static final GraphConfiguration CONFIG_BUILD_COUNT_TWO = createWithBuildCount(2);
+    private static final GraphConfiguration CONFIG_BUILD_DATE = createWithBuildDate();
+
+    private static final DateTime DAY = new DateTime(0, DateTimeZone.UTC).plusDays(1);
+    private static final DateTime PREVIOUS_DAY = DAY.minusDays(1);
+    private static final DateTime SAME_DAY = DAY.plusHours(4);
+    private static final DateTime NEXT_DAY = DAY.plusDays(1);
+
+    private static final StaticAnalysisRun RUN_PREVIOUS_DAY = createRun(1, PREVIOUS_DAY);
+    private static final StaticAnalysisRun RUN_DAY = createRun(2, DAY);
+    private static final StaticAnalysisRun RUN_SAME_DAY = createRun(3, SAME_DAY);
+    private static final StaticAnalysisRun RUN_NEXT_DAY = createRun(4, NEXT_DAY);
+
+    private static final List<Integer> FIRST_SERIES = series(0, 1, 2);
+    private static final List<Integer> SECOND_SERIES = series(3, 4, 5);
+    private static final List<Integer> FORTH_SERIES = series(9, 10, 11);
+    private static final List<Integer> AVERAGE_SECOND_AND_THIRD_SERIES = series(4, 5, 6);
+
+    private static Iterable<Object> createDataSetData() {
+        return asList(
+                new TestArgumentsBuilder()
+                        .setTestName("build count 0, 2 runs")
```

Comment:
3?

Reply:
Auf Grund der Antwort von oben muss ich glaube ich nochmal nachhaken: es werden ja 3 runs übergeben in Zeile 50, sollte der Text dann nicht auch 3 runs heißen? Oder sind das 2 skipped runs?